Scripts/gen_Primary.sh

```diff
@@ -146,45 +140,34 @@ else
   exit 1
 fi
 
-dataset=sim.mu2e.${TYPE}Stops${CAT}.${STOPS_SCAMPAIGN}.art
+dataset=sim.mu2e.${TYPE}Stops${CAT}.${STOPS_CAMPAIGN}.art
 
 if [[ "${TYPE}" == "Muminus" ]] ||  [[ "${TYPE}" == "Muplus" ]]; then
   resampler=TargetStopResampler
 elif [[ "${TYPE}" == "Piminus" ]] ||  [[ "${TYPE}" == "Piplus" ]]; then
   resampler=TargetPiStopResampler
-elif [[ "${TYPE}" == "Cosmic" ]]; then
-  dataset=sim.mu2e.${TYPE}DSStops${PRIMARY}.${stopsconf}.art
-  resampler=${TYPE}Resampler
 else
   resampler=${TYPE}StopResampler
 fi
 
-rm -f Stops.txt
-#samweb list-file-locations --schema=root --defname="$dataset"  | cut -f1 > Stops.txt
-samListLocations --defname $dataset > Stops.txt
+
+samweb list-definition-files $dataset  > Stops.txt
+
 # calucate the max skip from the dataset
 nfiles=`samCountFiles.sh $dataset`
 nevts=`samCountEvents.sh $dataset`
 let nskip=nevts/nfiles
 # write the template
 rm -f primary.fcl
 
-if [[ "${TYPE}" == "Cosmic" ]]; then
-  echo "#include \"Production/JobConfig/cosmic/S2Resampler${PRIMARY}.fcl\"" >> primary.fcl
-elif [[ "${TAG}" == "DIOtail" ]]; then
-  echo "#include \"Production/JobConfig/primary/DIOtail.fcl\"" >> primary.fcl
-else
-  echo "#include \"Production/JobConfig/primary/${PRIMARY}.fcl\"" >> primary.fcl
-fi
-
+echo "#include \"Production/JobConfig/primary/${PRIMARY}.fcl\"" >> primary.fcl
 echo physics.filters.${resampler}.mu2e.MaxEventsToSkip: ${nskip} >> primary.fcl
 echo "services.GeometryService.bFieldFile : \"${FIELD}\"" >> primary.fcl
+echo outputs.PrimaryOutput.fileName: \"dts.owner.${PRIMARY}.version.sequencer.art\"  >> primary.fcl
 
-if [[ "${TAG}" == "DIOtail" ]]; then
-  echo "DIOTail"
+if [[ "${PRIMARY}" == "DIOtail" ]]; then
   echo physics.producers.generate.decayProducts.spectrum.ehi: ${ENDMOM}        >> primary.fcl
   echo physics.producers.generate.decayProducts.spectrum.elow: ${STARTMOM}    >> primary.fcl
-  echo outputs.PrimaryOutput.fileName: \"dts.owner.${PRIMARY}.version.sequencer.art\"  >> primary.fcl
```

Comment:
I think I understand why this was encapsulated. Since in the DIOtail "primary name" becomes DIOtailp95MeVc etc. and we need to keep that part of the name in the dts file. So it needs to be within there. We need to retain the "tag" as DIOtail so we can reuse the .fcl input for both

Reply:
It doesnt matter which you choose but the point is the .fcl is multi purpose is can be used for any DIO startmom but the file name must have the start mom in there